lib/src/mocks/http.dart

```diff
@@ -156,10 +179,12 @@ class MockHttp {
 
 class _RequestExpectation {
   Object failWith;
+  final CaseInsensitiveMap headers;
   final String method;
   WResponse respondWith;
   final Uri uri;
 
-  _RequestExpectation(String this.method, Uri this.uri,
+  _RequestExpectation(
+      CaseInsensitiveMap this.headers, String this.method, Uri this.uri,
```

Comment:
Could we move `headers` to the third positional parameter? I know this is alphabetized, but with positional args, order matters, and method -> uri -> headers seems like the most logical ordering.

Also the type should be `Map<String, String>` here - it will be converted to a `CaseInsenstiveMap` internally